### ISSUE TITLE
`@media (color-gamut: ...)` checks for color fallbacks

### ISSUE THREAD

**Issue by romainmenke** [csstools/postcss-plugins#1049]
```css
.foo {
  color: oklch(...)
}

/* becomes */

.foo {
  color: rgb(...);
  color: color(display-p3 ...);
}
```

But this doesn't take into account the hardware capabilities of the current device.

Ideal output :

```css
.foo {
  color: oklch(...)
}

/* becomes */

.foo {
  color: rgb(...);
}

@media (color-gamut: display-p3) {
  .foo {
    color: color(display-p3 ...);
  }
}
```

----

This requires similar tricks as progressive custom properties.

**Comment by ai** [comment]
We can think about it as a polyfill, because current browsers violate CSSWG spec and do gamut correction in a wrong way.

So we will fix compatibility issue.

**Comment by romainmenke** [comment]
Exactly :)

For me this is very similar to [`postcss-progressive-custom-properties`](https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-progressive-custom-properties#postcss-progressive-custom-properties-)

```css
:root {
	/* fallback */
	--a-color: red;
	/* progressive enhancement */
	--a-color: oklch(40% 0.234 0.39 / var(--opacity-50));
}

/* becomes */

:root {
	--a-color: red;
}

@supports (color: oklch(0% 0 0)) {
	:root {
		--a-color: oklch(40% 0.234 0.39 / var(--opacity-50));
	}
}
```

But then not for CSS variables and with `@supports` but for all wide gamut colors and with `@media`.

**Comment by romainmenke** [comment]
Hi @ai,

This shipped in `postcss-preset-env` `v9.2.0` :

https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-gamut-mapping#readme

You can try it on the playground : https://preset-env.cssdb.org/playground/#eyJzb3VyY2UiOiJhOmhvdmVyIHtcbiAgY29sb3I6IG9rbGNoKDQwJSAwLjI2ODczNTQzNSAzNC41Njg2MjYpO1xufVxuXG5hID4gc3BhbiB7XG4gIGNvbG9yOiBjb2xvci1taXgoaW4gb2tsY2gsIGN5YW4sIGdyZWVuIDI1JSk7XG59IiwiY29uZmlnIjp7ImJyb3dzZXJzIjpbIj4gMC4yJSBhbmQgbm90IGRlYWQiXSwibWluaW11bVZlbmRvckltcGxlbWVudGF0aW9ucyI6MCwic3RhZ2UiOjIsImxvZ2ljYWwiOnsiaW5saW5lRGlyZWN0aW9uIjoibGVmdC10by1yaWdodCIsImJsb2NrRGlyZWN0aW9uIjoidG9wLXRvLWJvdHRvbSJ9LCJmZWF0dXJlcyI6eyJuZXN0aW5nLXJ1bGVzIjp7Im5vSXNQc2V1ZG9TZWxlY3RvciI6ZmFsc2V9fX19

**Comment by ai** [comment]
Amazing. I will tweet about it later.
